core/drivers/crypto/crypto_api/mac/mac.c

```diff
@@ -20,12 +20,13 @@ TEE_Result drvcrypt_mac_alloc_ctx(struct crypto_mac_ctx **ctx, uint32_t algo)
 
 	assert(ctx);
 
-	if (algo_id >= TEE_MAIN_ALGO_MD5 && algo_id <= TEE_MAIN_ALGO_SHA512) {
+	if (algo_id >= TEE_MAIN_ALGO_MD5 && algo_id <= TEE_MAIN_ALGO_SHA512)
 		mac_alloc = drvcrypt_get_ops(CRYPTO_HMAC);
+	else if (algo_id >= TEE_MAIN_ALGO_AES && algo_id <= TEE_MAIN_ALGO_DES3)
```

Comment:
`else` should be enough here. The `mac_alloc()` function pointer will check if the algorithm is OK later.

Reply:
Ok